src/Octoshift/Services/GithubApi.cs

```diff
@@ -423,11 +423,11 @@ public virtual async Task<(string State, string SourceOrgUrl, string TargetOrgNa
         }
     }
 
-    public virtual async Task<string> StartBbsMigration(string migrationSourceId, string orgId, string repo, string targetToken, string archiveUrl, string targetRepoVisibility = null)
+    public virtual async Task<string> StartBbsMigration(string migrationSourceId, string bbsRepoUrl, string orgId, string repo, string targetToken, string archiveUrl, string targetRepoVisibility = null)
     {
         return await StartMigration(
             migrationSourceId,
-            "https://not-used",  // source repository URL
+            bbsRepoUrl,  // source repository URL
             orgId,
             repo,
             "not-used",  // source access token
```

Comment:
The access token and metadata URL fields are optional in the GraphQL API now!  I know that these fields are mostly unrelated to your changes, but it would be great to switch them to `null` while you're here!  If you'd like, you are also welcome to file a tech debt issue for this suggestion instead and ship this PR without this suggestion :+1: 

The biggest lift would be to remove the `!` in the query in the GithubApi to for `accessToken` indicate that they're optional:

https://github.com/github/gh-gei/blob/cff148886221324c29ac76754d7244845909b0d0/src/Octoshift/Services/GithubApi.cs#L294-L294

There are various places in tests where we build a query, and we'll also want to remove the `!`.  This should be a simple find-and-replace.  Here's an example:

https://github.com/github/gh-gei/blob/cff148886221324c29ac76754d7244845909b0d0/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs#L889-L889

Then, for the BBS tests, we'd want to remove the "not used" variables:

https://github.com/github/gh-gei/blob/b709e7b962f110bc03801a9db256ead8db433932/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs#L877-L878

For the `metadataArchiveUrl` and `accessToken` values here...

https://github.com/github/gh-gei/blob/b709e7b962f110bc03801a9db256ead8db433932/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs#L934-L935

...we'll simply turn this into a hard-coded, nulled string.  There's a good example of that here:

https://github.com/github/gh-gei/blob/b709e7b962f110bc03801a9db256ead8db433932/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs#L938-L938

Reply:
I made these changes (with below diff) but now getting null reference exceptions. Any idea on how to best resolve that?

```
@dpmex4527 ➜ /workspaces/gh-gei (bbs-send-repo-url-for-selecting-fork-support) $ git diff
diff --git a/src/Octoshift/Services/GithubApi.cs b/src/Octoshift/Services/GithubApi.cs
index 323d543..ea0fe65 100644
--- a/src/Octoshift/Services/GithubApi.cs
+++ b/src/Octoshift/Services/GithubApi.cs
@@ -291,7 +291,7 @@ public class GithubApi
                     $continueOnError: Boolean!,
                     $gitArchiveUrl: String,
                     $metadataArchiveUrl: String,
-                    $accessToken: String!,
+                    $accessToken: String,
                     $githubPat: String,
                     $skipReleases: Boolean,
                     $targetRepoVisibility: String,
diff --git a/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs b/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs
index 76665b1..e447bc2 100644
--- a/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs
+++ b/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs
@@ -778,7 +778,7 @@ public class GithubApiTests
                     $continueOnError: Boolean!,
                     $gitArchiveUrl: String,
                     $metadataArchiveUrl: String,
-                    $accessToken: String!,
+                    $accessToken: String,
                     $githubPat: String,
                     $skipReleases: Boolean,
                     $targetRepoVisibility: String,
@@ -874,9 +874,6 @@ public class GithubApiTests
         const string gitArchiveUrl = "GIT_ARCHIVE_URL";
         const string targetToken = "TARGET_TOKEN";
 
-        const string unusedSourceToken = "not-used";
-        const string unusedMetadataArchiveUrl = "https://not-used";
-
         const string query = @"
                 mutation startRepositoryMigration(
                     $sourceId: ID!,
@@ -886,7 +883,7 @@ public class GithubApiTests
                     $continueOnError: Boolean!,
                     $gitArchiveUrl: String,
                     $metadataArchiveUrl: String,
-                    $accessToken: String!,
+                    $accessToken: String,
                     $githubPat: String,
                     $skipReleases: Boolean,
                     $targetRepoVisibility: String,
@@ -931,8 +928,8 @@ public class GithubApiTests
                 repositoryName = GITHUB_REPO,
                 continueOnError = true,
                 gitArchiveUrl,
-                metadataArchiveUrl = unusedMetadataArchiveUrl,
-                accessToken = unusedSourceToken,
+                metadataArchiveUrl = (string)null,
+                accessToken = (string)null,
                 githubPat = targetToken,
                 skipReleases = false,
                 targetRepoVisibility = (string)null,
```

```
@dpmex4527 ➜ /workspaces/gh-gei (bbs-send-repo-url-for-selecting-fork-support) $ dotnet test src/OctoshiftCLI.Tests/OctoshiftCLI.Tests.csproj --no-build --verbosity normal
Build started 5/25/2023 8:20:16 PM.
Test run for /workspaces/gh-gei/src/OctoshiftCLI.Tests/bin/Debug/net6.0/OctoshiftCLI.Tests.dll (.NETCoreApp,Version=v6.0)
Microsoft (R) Test Execution Command Line Tool Version 17.3.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
Unexpected response when retrieving User ID
{"invalid":{"PublicAlias":{"value":"foo-user-id"}}}
Description:
  Adds a team to a repo with a specific role/permission
  Note: Expects GH_PAT env variable or --github-pat option to be set.

Usage:
  add-team-to-repo [options]

Options:
  --github-org <github-org> (REQUIRED)
  --github-repo <github-repo> (REQUIRED)
  --team <team> (REQUIRED)
  --role <admin|maintain|pull|push|triage> (REQUIRED)  The only valid values are: pull, push, admin, maintain, triage. For more details see https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository-permissions, custom repository roles are not currently supported.
  --github-pat <github-pat>
  --verbose
  --version                                            Show version information
  -?, -h, --help                                       Show help and usage information


[xUnit.net 00:00:01.46]     OctoshiftCLI.Tests.Octoshift.Services.GithubApiTests.StartBbsMigration_Returns_New_Repository_Migration_Id [FAIL]
  Failed OctoshiftCLI.Tests.Octoshift.Services.GithubApiTests.StartBbsMigration_Returns_New_Repository_Migration_Id [11 ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at OctoshiftCLI.Services.GithubApi.StartMigration(String migrationSourceId, String sourceRepoUrl, String orgId, String repo, String sourceToken, String targetToken, String gitArchiveUrl, String metadataArchiveUrl, Boolean skipReleases, String targetRepoVisibility, Boolean lockSource) in /workspaces/gh-gei/src/Octoshift/Services/GithubApi.cs:line 352
   at OctoshiftCLI.Services.GithubApi.StartBbsMigration(String migrationSourceId, String bbsRepoUrl, String orgId, String repo, String targetToken, String archiveUrl, String targetRepoVisibility) in /workspaces/gh-gei/src/Octoshift/Services/GithubApi.cs:line 428
   at OctoshiftCLI.Tests.Octoshift.Services.GithubApiTests.StartBbsMigration_Returns_New_Repository_Migration_Id() in /workspaces/gh-gei/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:line 965
--- End of stack trace from previous location ---

Failed!  - Failed:     1, Passed:   777, Skipped:     0, Total:   778, Duration: 7 s - /workspaces/gh-gei/src/OctoshiftCLI.Tests/bin/Debug/net6.0/OctoshiftCLI.Tests.dll (net6.0)

Build FAILED.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:09.33
```

Reply:
You know, I was working on this myself to help you resolve these test failures, and it turned out to be quite a bit more work than I expected :smile: Let's hold off on this for now and remove the "not used" parameters in a follow-up PR :+1:

Reply:
Sounds good. I've added this as a cleanup task in https://github.com/github/gh-gei/issues/1004